pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java

```diff
@@ -77,17 +79,25 @@ public Consumer subscribe() throws PulsarClientException {
 
     @Override
     public CompletableFuture<Consumer> subscribeAsync() {
-        if (topicName == null) {
+        if (topicName == null && topicNames == null) {
             return FutureUtil
-                    .failedFuture(new IllegalArgumentException("Topic name must be set on the producer builder"));
+                    .failedFuture(new IllegalArgumentException("Topic name must be set on the consumer builder"));
+        }
+        if (topicName != null && topicNames != null) {
+            return FutureUtil
+                .failedFuture(new IllegalArgumentException("Should set either topicName or topicNames not both on the consumer builder"));
         }
 
         if (subscriptionName == null) {
             return FutureUtil.failedFuture(
-                    new IllegalArgumentException("Subscription name must be set on the producer builder"));
+                    new IllegalArgumentException("Subscription name must be set on the consumer builder"));
         }
 
-        return client.subscribeAsync(topicName, subscriptionName, conf);
+        if (topicName != null) {
+            return client.subscribeAsync(topicName, subscriptionName, conf);
+        } else {
+            return client.subscribeAsync(topicNames, subscriptionName, conf);
```

Comment:
@zhaijack I think we should remove the `subscribeAsync(List<String> topics)` method because all the `subscribe()` methods are deprecated in 2.0

Reply:
Oh, Thanks, I see.